packages/kit/src/runtime/client/client.js

```diff
@@ -1513,6 +1528,54 @@ export function create_client({ target, base }) {
 				});
 			});
 
+			target.addEventListener('submit', (event) => {
+				if (event.defaultPrevented) return;
+
+				const form = /** @type {HTMLFormElement} */ (
+					HTMLFormElement.prototype.cloneNode.call(event.target)
+				);
+
+				const submitter = /** @type {HTMLButtonElement | HTMLInputElement} */ (event.submitter);
+
+				const method = event.submitter?.hasAttribute('formmethod')
+					? submitter.formMethod
+					: form.method;
+
+				if (method !== 'get') return;
+
+				const url = new URL(
+					event.submitter?.hasAttribute('formaction') ? submitter.formAction : form.action
+				);
+
+				if (is_external_url(url, base)) return;
+
+				const { noscroll, reload } = get_router_options(
+					/** @type {HTMLFormElement} */ (event.target)
```

Comment:
```suggestion
					/** @type {HTMLFormElement | null} */ (event.target)
```

again, probably shouldn't pretend nullable things aren't possibly null and should just handle null in the function or return early

Reply:
this is just TypeScript being dumb though, no? Could `event.target` ever actually be `null`?

Reply:
It's not null in this case though, we know it exists. The types don't, that's why they play it safe.